Packs/Code42/Integrations/Code42/Code42.py

```diff
@@ -194,246 +208,294 @@ def build_query_payload(args):
     Build a query payload combining passed args
     """
     search_args = []
-    if args.get('hash'):
-        if len(args['hash']) == 32:
-            search_args.append(MD5.eq(args['hash']))
-        elif len(args['hash']) == 64:
-            search_args.append(SHA256.eq(args['hash']))
-    if args.get('hostname'):
-        search_args.append(OSHostname.eq(args['hostname']))
-    if args.get('username'):
-        search_args.append(DeviceUsername.eq(args['username']))
-    if args.get('exposure'):
+    if args.get("hash"):
+        if len(args["hash"]) == 32:
+            search_args.append(MD5.eq(args["hash"]))
+        elif len(args["hash"]) == 64:
+            search_args.append(SHA256.eq(args["hash"]))
+    if args.get("hostname"):
+        search_args.append(OSHostname.eq(args["hostname"]))
+    if args.get("username"):
+        search_args.append(DeviceUsername.eq(args["username"]))
+    if args.get("exposure"):
         # Because the CLI can't accept lists, convert the args to a list if the type is string.
-        if isinstance(args['exposure'], str):
-            args['exposure'] = args['exposure'].split(',')
-        search_args.append(ExposureType.is_in(args['exposure']))
+        if isinstance(args["exposure"], str):
+            args["exposure"] = args["exposure"].split(",")
+        search_args.append(ExposureType.is_in(args["exposure"]))
     # Convert list of search criteria to *args
     query = FileEventQuery.all(*search_args)
-    query.page_size = args.get('results')
-    LOG('File Event Query: {}'.format(query))
+    query.page_size = args.get("results")
+    LOG("File Event Query: {}".format(query))
     return str(query)
 
 
 @logger
 def map_observation_to_security_query(observation, actor):
     file_categories: Dict[str, Any]
-    observation_data = json.loads(observation['data'])
+    observation_data = observation["data"]
     search_args = []
     exp_types = []
-    exposure_types = observation_data['exposureTypes']
-    begin_time = observation_data['firstActivityAt']
-    end_time = observation_data['lastActivityAt']
-    if observation['type'] == 'FedEndpointExfiltration':
+    exposure_types = observation_data["exposureTypes"]
+    begin_time = observation_data["firstActivityAt"]
+    end_time = observation_data["lastActivityAt"]
+    if observation["type"] == "FedEndpointExfiltration":
         search_args.append(DeviceUsername.eq(actor))
     else:
         search_args.append(Actor.eq(actor))
-    search_args.append(EventTimestamp.on_or_after(
-        int(time.mktime(time.strptime(begin_time.replace('0000000', '000'), "%Y-%m-%dT%H:%M:%S.000Z")))))
-    search_args.append(EventTimestamp.on_or_before(
-        int(time.mktime(time.strptime(end_time.replace('0000000', '000'), "%Y-%m-%dT%H:%M:%S.000Z")))))
+
+    begin = time.mktime(time.strptime(begin_time.replace("0000000", "000"), "%Y-%m-%dT%H:%M:%S.000Z"))
```

Comment:
I think all this ends up doing is truncating 3 of the zeros and making it an int again, we do this in a much simpler way in the event extractor